src/Grpc.AspNetCore.Web/Internal/Base64PipeReader.cs

```diff
@@ -158,31 +170,57 @@ public async override ValueTask<ReadResult> ReadAsync(CancellationToken cancella
             }
 
             // Limit result to complete base64 segments (multiples of 4)
-            var buffer = innerResult.Buffer.Slice(0, (innerResult.Buffer.Length / 4) * 4);
+            var newResultLength = innerResult.Buffer.Length - _currentInnerBufferRead;
+            var newResultValidLength = (newResultLength / 4) * 4;
```

Comment:
Curious if bitmasking would be faster than a division and a multiplication here. Not sure if the compiler is smart enough to automatically do that. Also, I would assume the effect here would be pretty small anyways, but I was just curious.

Reply:
Anything here will be dwarfed by the calls to ReadAsync and base64 decoding.